File: 168_easy_string_index/Dart/main.dart
#!/usr/bin/env dart


final List<String> TEST_CASES = [
    r"""lazy fox jumps over a dog""",
    r"""easy does it hello world""",
    r"""...You...!!!@!3124131212 Hello have this is a --- string Solved !!...? to test @\n\n\n#!#@#@%$**#$@ Congratz
this!!!!!!!!!!!!!!!!one ---Problem\n\n"""

];


String getWord(List<String> words, int index) {
  if (index >= 0 && index < words.length) {
    return words[index];
  } else {
    return "";
  }
}

void solveTestCase(String input, [List<int> indexes = const [-10, -1, 0, 1, 2, 3, 7, 100, 1000]]) {
  print('solveTestCase START for input ' + input);
  print(input);

  List<String> words = input.split(new RegExp(r'[^a-zA-Z0-9]+'));

  for (var idx in indexes) {
    print('Word at index $idx: ${getWord(words, idx)}');
  }

}


void main() {

  int i = 0;
  for (String testCase in TEST_CASES) {
    print('==== Processing case $i');
    solveTestCase(testCase);
    i++;
  }
}

File: 168_easy_string_index/C#/main.cs
#define CONTRACTS_FULL
using System;
using System.Diagnostics.Contracts;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using System.Text.RegularExpressions;

namespace ConsoleApplication2
{

    class IndexedString
    {
        private string[] _words;

        public string this[int key]
        {
            get
            {
                if (key < 0 || key >= _words.Length)
                {
                    return "";

                }
                else
                {
                    return _words[key];

                }

            }

        }

        public IndexedString(string input)
        {
            var regex = new Regex("[^A-Za-z]+");
            _words = regex.Split(input);

        }

    }


    class Program
    {


        static void Main(string[] args)
        {
            Console.WriteLine("Hello world ");


            var a = new IndexedString("hello world ");

            for (int i = 0; i < 10; i++)
            {
                Console.WriteLine(i + ": " + a[i]);
            }


            var b = new IndexedString("...You...!!!@!3124131212 Hello have this is a --- string Solved " + 
                "!!...? to test @\n\n\n#!#@#@%$**#$@ Congratz this!!!!!!!!!!!!!!!!one ---Problem\n\n");

            for (int i = 0; i < 20; i++)
            {
                Console.WriteLine(i + ": " + b[i]);

            }

            Console.WriteLine("Press any key to continue ... ");
            Console.ReadKey();
        }

    }
}
